### PR TITLE
Add page for typescript client generator

### PR DESCRIPTION
**Pull request by luca-nardelli**
As discussed in https://github.com/api-platform/api-platform/issues/930

### CODE REVIEW

File: client-generator/typescript.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+...or globally
+
+    $ npm install -g @api-platform/client-generator
+
+(If you are using `yarn`, use `yarn add` or `yarn global add`)
+
+Then, in the app directory, generate the files for the resource you want:
+
+    $ generate-api-platform-client -g typescript https://demo.api-platform.com src/ --resource foo
```
Comment (dunglas) [inline client-generator/typescript.md]: You could use `npx` directly.

File: client-generator/typescript.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    # Replace the URL by the entrypoint of your Hydra-enabled API
+    # "src/" represents where the interfaces will be generated
+    # Omit the resource flag to generate files for all resource types exposed by the API
+
+This command parses the Hydra documentation and creates one `.ts` file for each Api Resource you have defined in your application, in the `interfaces` subfolder.
```
Comment (dunglas) [inline client-generator/typescript.md]:
```suggestion
This command parses the Hydra documentation and creates one `.ts` file for each API Resource you have defined in your application, in the `interfaces` subfolder.
```

File: client-generator/typescript.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Typescript interfaces
+
+The Typescript Generator allows you to create typescript interfaces that you can embed in any Typescript-enabled project (Angular, Vue, etc...)
```
Comment (dunglas) [inline client-generator/typescript.md]:
```suggestion
The TypeScript Generator allows you to create [TypeScript interfaces](https://www.typescriptlang.org/docs/handbook/interfaces.html) that you can embed in any TypeScript-enabled project (React, Vue.js, Angular..)
```

File: client-generator/typescript.md (hunk 1 above)
Comment (luca-nardelli) [reply client-generator/typescript.md]: Should I write it as an alternative way to invoke the command or should we replace the whole installation part in favor of `npx`?

Comment (dunglas) [reply client-generator/typescript.md]: I would replace it.

Comment (luca-nardelli) [reply client-generator/typescript.md]: Ok, I'll update it. 

By the way, I've tried to run the command through `npx` on my machine but it doesn't work.
```
luca@luca-mipro:~/Work/Personal/api-platform$ npx @api-platform/client-generator
npx: installed 284 in 10.132s
Cannot find module 'lodash'
```
However, the globally installed version works.

I'll open an issue in the client-generator repo to see if this is an issue that affects others or just me.

**Comment by luca-nardelli** [comment]
If everything's good, I'll squash the commits into one

File: client-generator/typescript.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Typescript interfaces
```
Comment (dunglas) [inline client-generator/typescript.md]:
```suggestion
# Typescript Interfaces
```

**Comment by dunglas** [comment]
Thanks @luca-nardelli!

**Comment by luca-nardelli** [comment]
You're welcome!

**Pull request merged**